### PR TITLE
Enable python 3.8 test using devops image for 3.8

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Python 3.8 is available now in Azure pipelines. This PR is to add Python 3.8 into test matrix.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: eng/pipelines/templates/jobs/archetype-sdk-nightly.yml [hunk 1]
```diff
@@ -100,32 +105,3 @@ jobs:
...
-    dependsOn:
-       - 'Build'
-
-    timeoutInMinutes: 300
-
-    pool:
-      vmImage: 'ubuntu-16.04'
-
-    steps:
-    - template: ../steps/test-nightly-python38.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-nightly.yml]: Should we delete `../steps/test-nightly-python38.yml` as well?

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 2]
```diff
@@ -104,35 +109,6 @@ jobs:
               artifactName: 'artifacts' 
               targetPath: $(Build.ArtifactStagingDirectory)
 
-  - job: 'Test_Linux_Python38'
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml]: Do we think we need to have a "nightly" for the next version past 3.8 to try and keep up with the latest versions?

File: eng/pipelines/templates/jobs/archetype-sdk-nightly.yml (hunk 1 above)
Comment (praveenkuttappan) [reply eng/pipelines/templates/jobs/archetype-sdk-nightly.yml]: yes. I thought of submitting separate PR to delete them once this change is successfully merged. Please let me know if you want me to include that change in this PR itself.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml (hunk 2 above)
Comment (praveenkuttappan) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml]: As discussed, we have an edge python pipeline to test next version of python.

**Pull request merged**